### PR TITLE
Management osgi test fixes

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

**Comment by anthonydahanne** [comment]
tests aren't passing ?

File: core/src/main/java/org/ehcache/core/internal/statistics/DefaultCacheStatistics.java [hunk 1]
```diff
@@ -91,16 +96,16 @@ public DefaultCacheStatistics(InternalCache<?, ?> cache) {
...
     Map<String, org.terracotta.statistics.ValueStatistic<?>> knownStatistics = new HashMap<>(30);
-    knownStatistics.put("Cache:HitCount", counter(this::getCacheHits));
-    knownStatistics.put("Cache:MissCount", counter(this::getCacheMisses));
-    knownStatistics.put("Cache:PutCount", counter(this::getCachePuts));
-    knownStatistics.put("Cache:RemovalCount", counter(this::getCacheRemovals));
-    knownStatistics.put("Cache:EvictionCount", counter(this::getCacheEvictions));
-    knownStatistics.put("Cache:ExpirationCount", counter(this::getCacheExpirations));
-
-    for (TierStatistics tier : tierStatistics.values()) {
+    knownStatistics.put("Cache:HitCount", ValueStatistics.counter(this::getCacheHits));
```
Comment (AbfrmBlr) [inline core/src/main/java/org/ehcache/core/internal/statistics/DefaultCacheStatistics.java]: Can we avoid this changeset by having an static import?

**Comment by AbfrmBlr** [comment]
Most of it is refactoring which is fine by me, but I want to know why we added ExtendedStatisticsService?

**Comment by chrisdennis** [comment]
- [ ] does this block things

**Comment by cljohnso** [comment]
@chrisdennis Re-base this and I'll take a crack at reviewing it.

**Comment by chrisdennis** [comment]
Should already be rebased.

**Review by cljohnso — COMMENTED** [review]
Not sure if anything I've noted warrants action ...

File: core/src/main/java/org/ehcache/core/internal/statistics/DefaultStatisticsService.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    return stats;
+  }
+
+  @Override
+  public void registerWithParent(Object toAssociate, Object parent) {
+    StatisticsManager.associate(toAssociate).withParent(parent);
+  }
+
+  @Override
+  public <K, V, S extends Enum<S>, T extends Enum<T>> org.ehcache.core.statistics.OperationStatistic<T> registerStoreStatistics(Store<K, V> store, String targetName, int tierHeight, String tag, Map<T, Set<S>> translation, String statisticName) {
```
[the remark this answers is not in the document]
Comment (chrisdennis) [reply core/src/main/java/org/ehcache/core/internal/statistics/DefaultStatisticsService.java:76]: Probably used to have a clash with the type in the stats library with the same name - it doesn't any more so I added an import.

File: core/src/main/java/org/ehcache/core/internal/statistics/DefaultStatisticsService.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+public class DefaultStatisticsService implements StatisticsService, CacheManagerListener {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultStatisticsService.class);
+
+  private final ConcurrentMap<String, DefaultCacheStatistics> cacheStatistics = new ConcurrentHashMap<>();
+
+  private volatile InternalCacheManager cacheManager;
+  private volatile boolean started = false;
+
+  public CacheStatistics getCacheStatistics(String cacheName) {
```
[the remark this answers is not in the document]
Comment (chrisdennis) [reply core/src/main/java/org/ehcache/core/internal/statistics/DefaultStatisticsService.java:62]: Yes.

File: core/src/main/java/org/ehcache/core/internal/statistics/DefaultStatisticsService.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  /**
+   * From the Map of translation, we extract one of the items to get the declaring class of the enum.
+   *
+   * @param translation translation map
+   * @param <S> type of the outcome
+   * @param <T> type of the possible translations
+   * @return the outcome type
+   */
+  private static <S extends Enum<S>, T extends Enum<T>> Class<S> getOutcomeType(Map<T, Set<S>> translation) {
+    Map.Entry<T, Set<S>> first = translation.entrySet().iterator().next();
```
[the remark this answers is not in the document]
Comment (chrisdennis) [reply core/src/main/java/org/ehcache/core/internal/statistics/DefaultStatisticsService.java:100]: If it doesn't then the type isn't extractable and the `NoSuchElementException` is a good enough failure mode.

**Review by cljohnso — APPROVED** [review]
Approved with question ....

**Review by cljohnso — APPROVED** [review]

**Pull request merged**